featurebyte/api/catalog_get_by_id_mixin.py

```diff
@@ -460,6 +461,31 @@ def get_user_defined_function_by_id(
         """
         return UserDefinedFunction.get_by_id(id=id)
 
+    @update_and_reset_catalog
```

Comment:
oh whats this annotation

Reply:
it's because our catalog id is stored in a global state variable, but we often times make catalog specific calls. so sometimes we need to update temporarily update the global state to the calling catalogs state.

for example,
```
catalog_a = Catalog.create("a")
catalog_b = Catalog.create("b")
use_catalog("b")  # updates global state to B

catalog_a.list_targets()
# we expect to see the targets of catalog A, but our global state is in B. 
# as such, the annotation will temporarily set global catalog state to A, make the call to list_targets, then reset it back to B
```